src/app/clusters/thread-network-diagnostics-server/thread-network-diagnostics-server.cpp

```diff
@@ -132,6 +136,62 @@ CHIP_ERROR ThreadDiagosticsAttrAccess::Read(const ConcreteReadAttributePath & aP
     return CHIP_NO_ERROR;
 }
 
+class ThreadDiagnosticsDelegate : public DeviceLayer::ThreadDiagnosticsDelegate
+{
+    // Notified when the Node’s connection status to a Thread network has changed.
+    void OnConnectionStatusChanged(ConnectionStatusEnum newConnectionStatus) override
+    {
+        MATTER_TRACE_SCOPE("OnConnectionStatusChanged", "ThreadDiagnosticsDelegate");
```

Comment:
That works but can't the app layer already register to the posted Event by the ThreadStackManager if need be?

Reply:
Well sure, I think it's already done by majority of applications. To be honest I've just copied existing Wi-Fi solution and thought it may be useful in some cases for Thread as well. But from what you wrote I think it's sort of redundant and can be removed until some other use case appears.